src/tools/pkcs11-tool.c

```diff
@@ -5494,8 +5547,16 @@ static void show_dobj(CK_SESSION_HANDLE sess, CK_OBJECT_HANDLE obj)
 		printf(" private");
 	if (!getMODIFIABLE(sess, obj) && !getPRIVATE(sess, obj))
 		printf("<empty>");
+	printf("\n");
 
-	printf ("\n");
+	get_token_info(opt_slot, &info);
+	printf("  uri:            %s", get_uri(&info));
+	if ((label = getLABEL(sess, obj, NULL)) != NULL) {
+		const char *pelabel = percent_encode((unsigned char *)label, strlen(label));
+		printf(";object=%s", pelabel);
+		free(label);
+	}
```

Comment:
> Can you reuse the `label` variable from the previous call to `getLABEL`? These query the card/module directly for the value so I would prefer to avoid repetitive calls to `getLABEL` if possible.

I tried it, but, for some reason, it seemed to be failing (valgrind tests were failing). I don't know how this getWHATEVER mechanism works, but it seems that the free has to be done after the get.

Reply:
Search for the `getLABEL` (should be mentioned in some comment) in the `pkcs11-tool` source code its macro calling `C_GetAttributeValue`, allocating the value so you need to free it, but it can be done only once after you write the URI.

Reply:
As you can see, label re-usage results on valgrind errors (I don't understand exactly why):
https://github.com/OpenSC/OpenSC/actions/runs/9062510878/job/24896538180?pr=3130